feature_engine/variable_manipulation.py

```diff
@@ -119,15 +133,18 @@ def _find_or_check_categorical_variables(
 
     if variables is None:
```

Comment:
We only want to check and exclude potential datetime variables when `variables is None`.

When the user passes a variable list, I am inclined to go with the user's judgement. If they pass a datetime variable to be treated as categorical, I would let them do that. They hopefully do that for a reason. Maybe it is just a datetime with few dates, like easter, christmas or something else, and it could be treated as categorical.